src/benchmarklib/benchmark_runner.cpp

```diff
@@ -100,6 +100,8 @@ void BenchmarkRunner::run() {
   if (_config.output_file_path) {
     if (!_config.verify && !_config.enable_visualization) {
       std::ofstream output_file(*_config.output_file_path);
+      _listener_reports = nlohmann::json{};
+      _notify_listeners(Event::CreateReport);
```

Comment:
Why is `_listener_reports` a member but `listener_payload` a parameter? I like the second approach better.

Reply:
We now could add the listener reports to the payload for the `ItemFinished` event, as part of a tuple (using `std::any` as payload type). That would still mean passing it (the report only) as JSON, but since it will be used to serialize later that is okay?

Reply:
I don't fully understand what would be passed as a JSON. Do you mean the data that is written into the result JSON? Sure, that's a different thing.

Reply:
Yes. Just checking :)